sql-plugin/src/main/spark350/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2Exec.scala

```diff
@@ -192,14 +195,17 @@ trait GpuWritingSparkTask[W <: DataWriter[ColumnarBatch]] extends Logging with S
     // write the data and commit this writer.
     Utils.tryWithSafeFinallyAndFailureCallbacks(block = {
       while (iter.hasNext) {
-        if (count % CustomMetrics.NUM_ROWS_PER_UPDATE == 0) {
-          CustomMetrics.updateMetrics(dataWriter.currentMetricsValues, customMetrics)
-        }
-
         // Count is here.
         val batch = iter.next()
+        val numRows = batch.numRows
         write(dataWriter, batch)
-        count += batch.numRows
+
+        val lastCount = count
+        count += numRows
+
+        if ((numRows + (lastCount % NUM_ROWS_PER_UPDATE)) >= NUM_ROWS_PER_UPDATE) {
```

Comment:
No need to update the metric based on this 'NUM_ROWS_PER_UPDATE'. (I know CPU uses it , but it is for better performance by packing multiple rows for one single update.) On GPU, a batch already does the rows packing for us, so we can update the metric for each batch. And the update at line 211 is not necessary. 
e.g.
```
      while (iter.hasNext) {
        // Count is here.
        val batch = iter.next()
        write(dataWriter, batch)
        count += batch.numRows
        CustomMetrics.updateMetrics(dataWriter.currentMetricsValues, customMetrics)
      }
```